src/ansys/fluent/core/quantity.py

```diff
@@ -1,1369 +1,869 @@
+        return self.__sub__(__value)
+
+    def __neg__(self):
+        return Quantity(-self.value, self.unit_str)
+
+    def __gt__(self, __value):
+        self._arithmetic_precheck(__value)
+        return float(self) > float(__value)
+
+    def __ge__(self, __value):
+        self._arithmetic_precheck(__value)
+        return float(self) >= float(__value)
+
+    def __lt__(self, __value):
+        self._arithmetic_precheck(__value)
+        return float(self) < float(__value)
+
+    def __le__(self, __value):
+        self._arithmetic_precheck(__value)
+        return float(self) <= float(__value)
+
+    def __eq__(self, __value):
+        self._arithmetic_precheck(__value)
+        return float(self) == float(__value)
+
+    def __neq__(self, __value):
+        self._arithmetic_precheck(__value)
+        return float(self) != float(__value)
+
+
+class QuantityError(ValueError):
+    """Custom quantity errors."""
+
+    def __init__(self, from_unit, to_unit):
         """
-        Sum of powers.
+        Quantity errors with custom messages.
 
         Parameters
         ----------
-        base: str
-            Unit term
-        unit_str: str
+        from_unit: str
             Unit of quantity
+        to_unit: str
+            Desired conversion unit
+        """
+        self.from_unit = from_unit
+        self.to_unit = to_unit
+
+    def __str__(self):
+        return f"'{self.from_unit}' and '{self.to_unit}' have incompatible dimensions."
+
+
+class UnitsTable(object):
+    """Initializes a UnitsTable object with all table values and unit string manipluation methods.
+
+    Properties
+    ----------
+    api_quantity_map : dict
+        Quantity map values from settings API.
+    fundamental_units : dict
+        Fundamental units and properties representing Mass, Length, Time, Current, Chemical Amount, Light, Solid Angle, Angle, Temperature and Temperature Difference.
+    derived_units : dict
+        Derived units and properties composed of fundamental units.
+    multipliers : dict
+        Multiplier prefixes and respective factors.
+
+    Methods
+    -------
+    filter_unit_term()
+        Separate multiplier, base, and power from a unit term.
+    si_data()
+        Compute the SI unit string, SI multiplier, and SI offset from a unit string of any type.
+    si_converion()
+        Perform SI conversion based on quantity type.
+    condense()
+        Condenses a unit string by collecting like-terms.
+    get_type()
+        Returns the type associated with a unit string.
+
+    Returns
+    -------
+    UnitsTable instance.
+    """
+
+    def __init__(self):
+        self._data: dict = self._get_data()
+        self._api_quantity_map: dict = self._data["api_quantity_map"]
+        self._fundamental_units: dict = self._data["fundamental_units"]
+        self._derived_units: dict = self._data["derived_units"]
+        self._multipliers: dict = self._data["multipliers"]
+
+    def _get_data(self) -> dict:
```

Comment:
`get_data` may as well be a function outside of the `UnitsTable` class, it is not doing anything specific to `UnitsTable` and makes no use of the object with `self`. Could be moved and renamed to `get_json_data` or something similar.

Alternatively, maybe just put the contents of `get_data` inside `__init__()` if it is only used there?

Reply:
Update `UnitsTable` to get data from config file during `__init__`

https://github.com/ansys/pyfluent/blob/d785d42ec4c3fe35fd1c982c9b1c1d291f28eb6d/src/ansys/fluent/core/quantity/units_table.py#L27-L39